### PR TITLE
`Development`: Enable cypress reruns

### PR DESCRIPTION
**Pull request by heddendorp**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

To combat test flakiness retries in tests are a very easy and mostly cheap solution. This enables 2 retries for cypress tests.

### Description

Enables retries in cypress tests.   
Also adds logs about retries so we can measure to occurrence of flaky tests => includes retries and is green.   
Should increase test reliability without much overhead.

### Example log
```
────────────────────────────────────────────────────────────────────────────────────────────────────

  Running:  init/ImportUsers.cy.ts                                                         (1 of 24)
 LOG:   RERUN: (Attempt 1 of 3) Logs in once with all required users
 LOG:   RERUN: (Attempt 2 of 3) Logs in once with all required users
<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="Mocha Tests" time="4.095" tests="1" failures="0">
  <testsuite name="Root Suite" timestamp="2022-12-29T11:22:58" tests="1" file="init\ImportUsers.cy.ts" time="4.091" failures="0">
    <testcase name="Logs in once with all required users" time="0.657" classname="Logs in once with all required users">
    </testcase>
  </testsuite>
</testsuites>

  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     4 seconds                                                                        │
  │ Spec Ran:     init/ImportUsers.cy.ts                                                           │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
```

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test

### Suggestion
Change bamboo plan to include a retried label if `RERUN:` is found in the test log to help get an overview

### CODE REVIEW

File: src/test/cypress/support/index.ts [hunk 1]
```diff
@@ -56,4 +56,17 @@ Cypress.on('window:before:load', (win) => {
 Cypress.on('uncaught:exception', (err, runnable) => {
     return err.name === 'TypeError' && err?.stack?.includes('sentry');
 });
+
+/**
+ * We want to log out our test retries to the console to be able to see how often a test is retried.
```
Comment (ge65cer) [inline src/test/cypress/support/index.ts:61]:
```suggestion
 * We want to log our test retries to the console to be able to see how often a test is retried.
```

**Review by ge65cer — DISMISSED** [review]
Good idea!

File: src/main/docker/cypress/docker-compose.yml [hunk 2]
```diff
@@ -66,7 +66,7 @@ services:
             - CYPRESS_editorGroupName
             - CYPRESS_instructorGroupName
         # Wait up to 5 minutes until Artemis has booted
-        command: sh -c "cd /app/artemis/src/test/cypress && chmod 777 /root && npm ci && npm exec -- wait-on -i 1000 -t 300000 http://artemis-app:${SERVER_PORT} && npm run cypress:run"
+        command: sh -c "cd /app/artemis/src/test/cypress && chmod 777 /root && npm ci && npm exec -- wait-on -i 1000 -t 300000 http://artemis-app:${SERVER_PORT} && npm run cypress:run -- --config retries=2"
```
Comment (Strohgelaender) [inline src/main/docker/cypress/docker-compose.yml:69]: Is this `--` necessary? 
```suggestion
        command: sh -c "cd /app/artemis/src/test/cypress && chmod 777 /root && npm ci && npm exec -- wait-on -i 1000 -t 300000 http://artemis-app:${SERVER_PORT} && npm run cypress:run --config retries=2"
```

Comment (heddendorp) [reply src/main/docker/cypress/docker-compose.yml:69]: @Strohgelaender Yes, in my tries it showed that the `--` was still needed to pass the argument on to `cypress:run`. I too was under the impression this changed but think that might have been a yarn thing

File: src/test/cypress/support/index.ts (hunk 1 above)
Comment (heddendorp) [reply src/test/cypress/support/index.ts]: Thanks

**Review by Strohgelaender — APPROVED** [review]

**Review by ge65cer — APPROVED** [review]

**Review by TheZoker — APPROVED** [review]

**Review by simon-hng — APPROVED** [review]

**Pull request merged**